File: SeptemberCodingChallenge/Day-3-Repeated-Substring-Pattern.py
class Solution(object):
    def repeatedSubstringPattern(self, s):
        """
        :type s: str
        :rtype: bool
        """
        t = ""
        for x in range(0, len(s) / 2):
            t = t + s[x]
            l = len(t)
            for y in range(l, len(s), l):
                if s[y:(y + l)] == t:
                    if y < len(s) - l:
                        continue
                    else:
                        return True
                else:
                    break
        return False

File: SeptemberCodingChallenge/Day-3-Repeated-Substring-Pattern.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        String t = "";
        int len = s.length();
        for(int x = 0; x < (len / 2); x++){
            t = s.substring(0, x+1);
            int l = t.length();
            for(int y = l; y<=len-l ; y+=l){
                if(s.substring(y, y+l).equals(t)){
                    if(y < (len - l))
                        continue;
                    else
                        return true;}
                else
                    break;}}
        return false;
    }
}
